ui/src/components/Broker/InviteAcceptScreen.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react'
+
+import { useParty, useLedger } from '@daml/react'
+import { useWellKnownParties } from '@daml/dabl-react'
+import { BrokerInvitation } from '@daml.js/da-marketplace/lib/Marketplace/Broker'
+
+import { wrapDamlTuple } from '../common/damlTypes'
+import { ErrorMessage } from '../common/errorTypes'
+import { InviteAcceptTile, InviteAcceptButton, InviteTextField } from '../common/InviteAcceptTile'
+
+type Props = {
+    onLogout: () => void;
+}
+
+const InviteAcceptScreen: React.FC<Props> = ({ onLogout }) => {
+    const ledger = useLedger();
+    const issuer = useParty();
+    const operator = useWellKnownParties().userAdminParty;
+
+    const [ name, setName ] = useState<string>('')
+    const [ location, setLocation ] = useState<string>('')
+    const [ error, setError ] = useState<ErrorMessage>();
+    const [ loading, setLoading ] = useState(false);
+
+    async function submit() {
+        setLoading(true);
+        const key = wrapDamlTuple([operator, issuer]);
+        const args = { name, location };
+        await ledger.exerciseByKey(BrokerInvitation.BrokerInvitation_Accept, key, args)
+            .catch(err => console.error(err));
+        setLoading(false);
+    }
+
+    return (
+        <InviteAcceptTile
+            onLogout={onLogout}
+            role='Broker'
+            error={error}
+            setError={setError}
+        >
+            <InviteTextField
+                label='Name'
+                placeholder='The name of the Broker'
+                variable={name}
+                setter={setName}
+            />
+            <InviteTextField
+                label='Location'
+                placeholder='The location of the Broker'
+                variable={location}
+                setter={setLocation}
+            />
+            <InviteAcceptButton
+                loading={loading}
+                disabled={!name || !location}
+                submit={submit}
```

Comment:
without setting a `value` for each field the form clears on its own right? If so ill refactor my other work. Overlooked that.

Reply:
otherwise lets make sure it does clear

Reply:
What do you mean about the `value`? I don't think it would clear as is.

This form is only submitted once, and you can't submit unless all the fields are filled in. So I'm not sure it _needs_ to clear, but it may be better to do so in case there is a delay between the accept choice being exercised and the main page loading the registered contract where the user would see the filled out form and potentially be able to submit it twice.

 I can add the clearing to this and the other invitation forms.

Reply:
well in that case nevermind!